src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Http.Generators.StaticRouteHandlerModel.Emitters;
+internal static class EndpointParameterEmitter
+{
+    internal static string EmitSpecialParameterPreparation(this EndpointParameter endpointParameter)
+    {
+        return $"""
+                        var {endpointParameter.Name}_local = {endpointParameter.AssigningCode};
+""";
+    }
+
+    internal static string EmitQueryParameterPreparation(this EndpointParameter endpointParameter)
+    {
+        var builder = new StringBuilder();
+
+        // Preamble for diagnostics purposes.
+        builder.AppendLine($$"""
+                        // Endpoint Parameter: {{endpointParameter.Name}} (Type = {{endpointParameter.Type}}, IsOptional = {{endpointParameter.IsOptional}}, Source = {{endpointParameter.Source}})
+""");
+
+        // Grab raw input from HttpContext.
+        builder.AppendLine($$"""
+                        var {{endpointParameter.Name}}_raw = {{endpointParameter.AssigningCode}};
+""");
+
+        // If we are not optional, then at this point we can just assign the string value to the handler argument,
+        // otherwise we need to detect whether no value is provided and set the handler argument to null to
+        // preserve consistency with RDF behavior. We don't want to emit the conditional block to avoid
+        // compiler errors around null handling.
+        if (endpointParameter.IsOptional)
+        {
+            builder.AppendLine($$"""
+                        var {{endpointParameter.HandlerArgument}} = {{endpointParameter.Name}}_raw.Count > 0 ? {{endpointParameter.Name}}_raw.ToString() : null;
+""");
+        }
+        else
+        {
+            builder.AppendLine($$"""
+                        if (StringValues.IsNullOrEmpty({{endpointParameter.Name}}_raw))
```

Comment:
It looks like the pattern here is going to be slightly different than what we currently do in RDF. In RDF, we assign a `wasParamCheckFailure` variable to track if any binding failed then set the status code and return _after_ all the bindings have been attempted.

I'm fine with merging this as-is now, since I:

- Anticipate (hope?) that people aren't relying on side-effects from parameter-binding in their implementations
- I think we'll end up bringing back a `wasParamCheckFailure` variable as we add support for more binding here to keep the generated code more terse

Reply:
Yeah, I think once this is in, we can come up with a pattern for this. It is going to be "fun" replicating existing behaviors and quirks.